### PR TITLE
[IMP] product: Add demo data for multi-valued attribute

### PR DESCRIPTION
**Pull request by mathysPaul**
Adds demo data in the Product module to showcase the Multi-checkbox attribute. The 'Extra Options' attribute is introduced with the following choices:
- 1 year warranty extension (extra price 50)
- Cleaning kit (extra price 5)
- Protection kit (extra price 10)

This attribute is linked to the product:
- Outdoor dinning table:
  - Storable
  - Price : 589
  - Category: Outdoor furniture

This setup demonstrates the new attribute's capabilities in enhancing product customization without affecting variant generation.

task-3610666

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/product/data/product_demo.xml [hunk 1]
```diff
@@ -600,6 +626,21 @@
             <field name="image_1920" type="base64" file="product/static/img/product_product_d01-image.jpg"/>
         </record>
 
+        <record id="product_template_no_variant_attribute" model="product.template">
+            <field name="name">Outdoor dinning table</field>
+            <field name="categ_id" ref="product_category_outdoor_furniture"/>
+            <field name="list_price">589</field>
+            <field name="detailed_type">consu</field>
```
Comment (vava-odoo) [inline addons/product/data/product_demo.xml:633]: Default value
```suggestion
```

File: addons/product/data/product_demo.xml [hunk 2]
```diff
@@ -600,6 +626,21 @@
             <field name="image_1920" type="base64" file="product/static/img/product_product_d01-image.jpg"/>
         </record>
 
+        <record id="product_template_no_variant_attribute" model="product.template">
+            <field name="name">Outdoor dinning table</field>
```
Comment (vava-odoo) [inline addons/product/data/product_demo.xml:630]: Typo
```suggestion
            <field name="name">Outdoor dining table</field>
```

File: addons/product/data/product_demo.xml [hunk 3]
```diff
@@ -600,6 +626,21 @@
...
+            <field name="categ_id" ref="product_category_outdoor_furniture"/>
+            <field name="list_price">589</field>
+            <field name="detailed_type">consu</field>
+            <field name="image_1920" type="base64" file="product/static/img/product_template_no_va-image.png"/>
+        </record>
+
+        <!-- the product template attribute lines have to be defined before creating the variants -->
```
Comment (vava-odoo) [inline addons/product/data/product_demo.xml:637]: I don't think this comment is really useful, as it is already present in the file
```suggestion
```

File: addons/product/data/product_demo.xml [hunk 4]
```diff
@@ -600,6 +626,21 @@
...
+        <record id="product_template_no_variant_attribute_attribute_4_product_template_attribute_line" model="product.template.attribute.line">
+            <field name="product_tmpl_id" ref="product_template_no_variant_attribute"/>
+            <field name="attribute_id" ref="product_attribute_4"/>
+            <field name="value_ids" eval="[(6,0,[ref('product.product_attribute_value_7'), ref('product.product_attribute_value_8'), ref('product.product_attribute_value_9')])]"/>
```
Comment (vava-odoo) [inline addons/product/data/product_demo.xml:641]: No need to specify the module if it is defined in the same one in a `ref` + spaces after `,`
```suggestion
            <field name="value_ids" eval="[(6, 0, [ref('product_attribute_value_7'), ref('product_attribute_value_8'), ref('product_attribute_value_9')])]"/>
```

**Review by vava-odoo — CHANGES_REQUESTED** [review]
Thanks @mathysPaul 
Looks overall good, only made a few minor comments.
Could you please squash your commits?
Cheers!

File: addons/product/data/product_demo.xml [hunk 5]
```diff
@@ -34,6 +34,10 @@
             <field name="parent_id" ref="product_category_all"/>
             <field name="name">Home Construction</field>
         </record>
+        <record id="product_category_outdoor_furniture" model="product.category">
+            <field name="parent_id" ref="product_category_all"/>
```
Comment (vava-odoo) [inline addons/product/data/product_demo.xml:38]: Maybe a sub-category of Saleable?
```suggestion
            <field name="parent_id" ref="product_category_1"/>
```

File: addons/product/data/product_demo.xml [hunk 6]
```diff
@@ -600,6 +626,19 @@
...
         </record>
 
+        <record id="product_template_no_variant_attribute" model="product.template">
+            <field name="name">Outdoor dining table</field>
+            <field name="categ_id" ref="product_category_outdoor_furniture"/>
+            <field name="list_price">589</field>
+            <field name="image_1920" type="base64" file="product/static/img/product_template_no_va-image.png"/>
+        </record>
+
+        <record id="product_template_no_variant_attribute_attribute_4_product_template_attribute_line" model="product.template.attribute.line">
```
Comment (vava-odoo) [inline addons/product/data/product_demo.xml:636]: I know you followed the structure above, but this id looks too long to me.

**Review by vava-odoo — APPROVED** [review]
Hi @mathysPaul 
Thanks for the update. It looks good to me now. I only have two small remarks, to be discussed with the team.
I already approve so that it can move on.
Cheers!

**Review by Feyensv — APPROVED** [review]
@robodoo r+

Pushed updated commit to:
* rename xml ids to highlight more clearly what they represent
* update commit tag from ADD to IMP
* simplify template declaration, no need to have an xml id for the attribute line, the template can be created directly with the right configuration.

**Pull request closed without merging**